conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -269,6 +271,27 @@ def _get_exes(self, cpp_info, pkg_name, pkg_folder, pkg_folder_var):
 
         return exes
 
+    def _get_sources(self, cpp_info, pkg_name, pkg_folder, pkg_folder_var):
+        sources = {}
+
+        if cpp_info.has_components:
+            for name, comp in cpp_info.components.items():
+                if comp.sources:
+                    target_name = self._cmakedeps.get_property("cmake_target_name", self._conanfile,
+                                                               name)
+                    target = target_name or f"{pkg_name}::{name}"
+                    sources[target] = []
+                    for source in comp.sources:
+                        sources[target].append(self._path(source, pkg_folder, pkg_folder_var))
+        else:
+            if cpp_info.sources:
+                target_name = self._cmakedeps.get_property("cmake_target_name", self._conanfile)
+                target = target_name or f"{pkg_name}::{pkg_name}"
+                sources[target] = []
+                for source in cpp_info.sources:
+                    sources[target].append(self._path(source, pkg_folder, pkg_folder_var))
+        return sources
```

Comment:
Looking simpler? Only a suggestion:

```suggestion
    def _get_sources(self, cpp_info, pkg_name, pkg_folder, pkg_folder_var):
        def _fill_sources(info, comp_name=None):
            if info.sources:
                target_name = self._cmakedeps.get_property("cmake_target_name", self._conanfile,
                                                           comp_name=comp_name, check_type=list)
                target = target_name or f"{pkg_name}::{name}"
                sources[target].extend([self._path(source, pkg_folder, pkg_folder_var)
                                        for source in info.sources])

        sources = {}
        if cpp_info.has_components:
            for name, comp in cpp_info.components.items():
                _fill_sources(comp, comp_name=name)
        else:
            _fill_sources(cpp_info)
        return sources
```

Reply:
Had to revert the change as it was not working out of the box. When tweaking it I found it more complex to understand than the original code, so I reverted it.